src/activities/CreateMeasurement2D.ts

```diff
@@ -0,0 +1,117 @@
+
+}
+
+export interface CreateMeasurement2DOutputs {
+    measurement: __esri.AreaMeasurement2DViewModelMeasurement | __esri.DistanceMeasurement2DViewModelMeasurement | undefined;
+    measurementWidget: any,
+}
+
+/**
+ * @category ArcGIS Maps SDK for JavaScript
+ * @clientOnly
+ * @description Measure a distance or area on a 2D map.
+ * @helpUrl https://developers.arcgis.com/javascript/latest/sample-code/widgets-measurement-2d
+ * @supportedApps EXB, GWV
+ */
+@activate(MapProvider)
+export default class CreateMeasurement2D implements IActivityHandler {
+    async execute(
+        inputs: CreateMeasurement2DInputs,
+        context: IActivityContext,
+        type: typeof MapProvider
+    ): Promise<CreateMeasurement2DOutputs> {
+        const { linearUnit, measurementType, areaUnit } = inputs;
+        const areaUnits = ["metric", "imperial", "square-inches", "square-feet", "square-us-feet", "square-yards", "square-miles", "square-meters", "square-kilometers", "acres", "ares", "hectares"]
+        const linearUnits = ["metric", "imperial", "inches", "feet", "us-feet", "yards", "miles", "nautical-miles", "meters", "kilometers"]
+        const mapProvider = type.create();
+        await mapProvider.load();
+        if (!mapProvider.map) {
+            throw new Error("map is required");
+        }
+        if (!measurementType) {
+            throw new Error("measurementType is required");
+        }
+        if (["area", "distance"].indexOf(measurementType) == -1) {
+            throw new Error(`measurementType not supported: ${measurementType}`);
+        }
+        if (linearUnit && linearUnits.indexOf(linearUnit) == -1) {
+            throw new Error(`linear unit not supported: ${linearUnit}`);
+        }
+        if (areaUnit && areaUnits.indexOf(areaUnit) == -1) {
+            throw new Error(`area unit not supported: ${areaUnit}`);
+        }
+        const mapView = mapProvider.view as MapView;
+
+        /**
+         * Ideally this would be implemented using AreaMeasurement2DViewModel or DistanceMeasurement2DViewModel
+         * but there is an inconsistency in the Esri Widget View Model pattern with 
+         * measurements that forces the creation of a widget.  A standalone view model 
+         * does not initialize.
+         **/
+
+        let measurementWidget;
+
+        switch (measurementType) {
+            case "area":
+                measurementWidget = new AreaMeasurement2D({
+                    view: mapView,
+                    unit: areaUnit as any,
+                });
+                break;
+            case "distance":
+            default:
+                measurementWidget = new DistanceMeasurement2D({
+                    view: mapView,
+                    unit: linearUnit as any,
+                });
+                break;
+        }
+
+        measurementWidget.viewModel.start();
+        mapView.container.style.cursor = "crosshair";
```

Comment:
Setting the cursor can be trouble. We also aren't clearing it when we are done.

Reply:
removed.  It was an earlier idea.  I'm not a fan of not having any feedback to let the user know the tool is active but that does change once they click on the map.